ors-api/src/main/java/org/heigit/ors/api/controllers/SnappingAPI.java

```diff
@@ -0,0 +1,157 @@
+@RequestMapping("/v2/snap")
+@ApiResponse(responseCode = "400", description = "The request is incorrect and therefore can not be processed.")
+@ApiResponse(responseCode = "404", description = "An element could not be found. If possible, a more detailed error code is provided.")
+@ApiResponse(responseCode = "405", description = "The specified HTTP method is not supported. For more details, refer to the EndPoint documentation.")
+@ApiResponse(responseCode = "413", description = "The request is larger than the server is able to process, the data provided in the request exceeds the capacity limit.")
+@ApiResponse(responseCode = "500", description = "An unexpected error was encountered and a more detailed error code is provided.")
+@ApiResponse(responseCode = "501", description = "Indicates that the server does not support the functionality needed to fulfill the request.")
+@ApiResponse(responseCode = "503", description = "The server is currently unavailable due to overload or maintenance.")
+public class SnappingAPI {
+    static final CommonResponseEntityExceptionHandler errorHandler = new CommonResponseEntityExceptionHandler(SnappingErrorCodes.BASE);
+
+    private final SnappingService snappingService;
+
+    public SnappingAPI(SnappingService snappingService) {
+        this.snappingService = snappingService;
+    }
+
+    // generic catch methods - when extra info is provided in the url, the other methods are accessed.
+    @GetMapping
+    @Operation(hidden = true)
+    public void getGetMapping() throws MissingParameterException {
+        throw new MissingParameterException(SnappingErrorCodes.MISSING_PARAMETER, "profile");
+    }
+
+    @PostMapping
+    @Operation(hidden = true)
+    public String getPostMapping(@RequestBody SnappingApiRequest request) throws MissingParameterException {
+        throw new MissingParameterException(SnappingErrorCodes.MISSING_PARAMETER, "profile");
+    }
+
+    // Matches any response type that has not been defined
+    @PostMapping(value = "/{profile}/*")
+    @Operation(hidden = true)
+    public void getInvalidResponseType() throws StatusCodeException {
+        throw new StatusCodeException(HttpServletResponse.SC_NOT_ACCEPTABLE, SnappingErrorCodes.UNSUPPORTED_EXPORT_FORMAT, "This response format is not supported");
+    }
+
+    // Functional request methods
+    @PostMapping(value = "/{profile}")
+    @Operation(
+            description = """
+                    Returns a list of points snapped to the nearest edge in the graph. \
+                    This method does not accept any request body or parameters other than profile, locations, maximum_snapping_radius.\
+                    """,
+            summary = "Snapping Service"
+    )
+    @ApiResponse(
+            responseCode = "200",
+            description = "Standard response for successfully processed requests. Returns JSON.",
+            content = {@Content(
+                    mediaType = "application/geo+json",
```

Comment:
The response is not a geojson so this should be `application/json`. 
You can validate geojson with:
http://geojson.io/

See e.g. https://en.wikipedia.org/wiki/GeoJSON for geojson format.

Should we have that as response format as well?

Reply:
Done.